features/support/branch_helpers.rb

```diff
@@ -9,6 +9,20 @@ def branch_name_for_location location, branch
 end
 
 
+# Returns the location of the branch with the given name
+#
+# 'foo'          --> 'local'
+# 'origin/foo'   --> 'remote'
+# 'upstream/foo' --> 'upstream'
+def branch_location branch_name
+  case
+  when branch_name.start_with?('origin/') then 'remote'
+  when branch_name.start_with?('upstream/') then 'upstream'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.